### ISSUE TITLE
Modman and git submodules

### ISSUE THREAD

**Issue by devolonter** [blitz-research/monkey#21]
Yesterday I tested new module manager. It seems, everything works perfectly with all modules and repositories. But I’ve noticed that manager runs git clone without --recursive option. So, if module uses git submodules, directory with submodule will be empty after installing.

I’ve tried to just add the --recursive option, but it didn’t work. The installation was not completed, and I had to kill the git process manually... Although it works fine if I run the same command directly from the command line

I would like to know if there any chance that the git submodules will be also supported. Or  I   should refuse from having them in my repositories?

Thanks!

**Comment by blitz-research** [comment]
Will have a look at this eventually - do you have any immediate need for it?

Git does seem to start sub-processes up in a 'weird' way, enough to mean I couldn't cheat and just use shell commands with redirected stdout to drive everything. Which was probably for the best as it meant I had to write brl.process - which is probably where the issue is.

**Comment by devolonter** [comment]
No, it’s not urgent. At the moment I use submodules in flixel repository only for examples and 3rd party plugins. So, the module works fine without them.

Also, you can try to solve the issue with another command for getting submodules:

`git submodule update --init` after `git clone`

This command works not recursively.

P.S. Just for the information. Using submodules also require changes in the update method. You need to add run `git submodule update` after `git merge`
